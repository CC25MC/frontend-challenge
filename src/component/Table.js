import Table from "react-bootstrap/Table";

export const TableB = ({ data = [] }) => {
  return (
    <Table size="sm" striped bordered hover>
      <thead>
        <tr>
          <th>File Name</th>
          <th>Text</th>
          <th>Number</th>
          <th>Hex</th>
        </tr>
      </thead>
      <tbody>
        {data.length > 0 ? (
          data.map((i) => {
            return i.lines.map((item, key) => (
              <tr key={key}>
                <td>{item.file}</td>
                <td>{item.text}</td>
                <td>{item.number}</td>
                <td>{item.hex}</td>
              </tr>
            ));
          })
        ) : (
          <tr>
            <td colSpan={4} className={"text-center"}>There is no data!!!</td>
          </tr>
        )}
        {}
      </tbody>
    </Table>
  );
};
