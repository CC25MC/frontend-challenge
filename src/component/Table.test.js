import { render } from "@testing-library/react";

import { TableB } from "./Table";

describe("Table tests", () => {
  const data = [
    {
      file: "test2.csv",
      lines: [
        {
          file: "test2.csv",
          text: "vYISIpKVaG",
          number: 47661943,
          hex: "9999aae84d12a16846387e793eb4436c",
        },
      ],
    },
  ];

  it("Should render correctly", () => {
    const component = render(<TableB data={data} />);
    component.getByText(`${data[0].file}`);
    expect(component.container).toMatchSnapshot();
  });

  it("Should not show the data when", () => {
    const component = render(<TableB data={[]} />);
    component.getByText("There is no data!!!");
    expect(component.container).toMatchSnapshot();
  });
});
