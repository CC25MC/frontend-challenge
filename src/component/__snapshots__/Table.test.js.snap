// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table tests Should not show the data when 1`] = `
<div>
  <table
    class="table table-sm table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          File Name
        </th>
        <th>
          Text
        </th>
        <th>
          Number
        </th>
        <th>
          Hex
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          class="text-center"
          colspan="4"
        >
          There is no data!!!
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Table tests Should render correctly 1`] = `
<div>
  <table
    class="table table-sm table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          File Name
        </th>
        <th>
          Text
        </th>
        <th>
          Number
        </th>
        <th>
          Hex
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          test2.csv
        </td>
        <td>
          vYISIpKVaG
        </td>
        <td>
          47661943
        </td>
        <td>
          9999aae84d12a16846387e793eb4436c
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
